src/lib/autoinstall/package_gpg_check_handler.rb

```diff
@@ -0,0 +1,105 @@
+module Yast
+  module PackageGpgCheckHandler
+    CHK_OK         = 0 # Signature is OK
+    CHK_NOTFOUND   = 1 # Signature type is unknown
+    CHK_FAIL       = 2 # Signature does not verify
+    CHK_NOTTRUSTED = 3 # Signature is OK but key is not trusted
+    CHK_NOKEY      = 4 # Public key is unavailable
+    CHK_ERROR      = 5 # File does not exist or can't be open
+
+    ABORT  = "A" # I'm not sure if we could just return true/false and let the callback
+                 # to convert it.
+    IGNORE = "I"
+
+    # This command will produce something which last line will be like:
+    # DSA/SHA1, Mon 05 Oct 2015 04:24:50 PM WEST, Key ID 9b7d32f2d50582e6
+    FIND_KEY_ID_CMD = "rpm -qi --qf "\
+      "\"%%|DSAHEADER?{%%{DSAHEADER:pgpsig}}:{%%|RSAHEADER?{%%{RSAHEADER:pgpsig}}:{(none}|}|\" "\
+      " -p %s"
+
+    def initialize(data)
+      @result  = data["CheckPackageResult"]
+      @package = data["Package"]
+      @path    = data["Localpath"]
+      @config  = get_addon_config(data["profile"], data["product"])
+    end
+
+    def run
+      result =
+        case @result
+        when CHK_OK
+          handle_ok
+        when CHK_NOTFOUND
+          handle_unsigned
+        when CHK_NOKEY
+          handle_unknown
+        when CHK_FAIL
+          handle_failed
+        when CHK_NOTTRUSTED
+          handle_nontrusted
+        when CHK_ERROR
+          handle_error
+        else
+          raise "Unknown GPG check result for #{@package}"
+        end
+      result ? IGNORE : ABORT
+    end
+
+    private
+
+    # Handles CHK_OK result
+    #
+    # Always returns true
+    #
+    # @return [String] true
+    def handle_ok
+      true
+    end
+
+    # Handles the situation where GPG key is not found (unknown type or missing)
+    #
+    # @return [String] true if success; otherwise false.
+    def handle_unsigned
+      @config["accept_unsigned_handling"] == true
```

Comment:
NP: I'd prefer attribute accessor, e.g. `attr_accessor :config` and use `config["..."]` instead of `@config`.